### PR TITLE
Enables scanners (with a delay) and a couple other consoles for blind people

### PR DESCRIPTION
**Pull request by KoboldCommando**
# About The Pull Request

This expands the ability for blind people to use various tools and machines.  I'm favoring a liberal approach.  Currently it's a bit inconsistent what you can use while blind and deaf, and I was happily doing medical duties except for one thing: I couldn't use health analyzers at all.

Now, blind people can use health analyzers, atmospheric analyzers, and slime scanners.  They have a slight delay (to read the data to you) and you or the target moving will break the interaction.  Advanced health analyzers are quicker to act.

I also allowed the use of advanced camera consoles based on the fact that you can use all other forms of camera console and remote viewing, and the shuttle designator by the same logic.  In my mind, it's the future, so we have various ways for the computers to read and communicate things nonvisually, but they're displayed visually in the game for ease.

Here's an example of someone browsing the web with a screen reader.  Imagine this combined with facial recognition and stronger AI, being fed into your earpiece.  That's how I imagine a blind person using the various computers on station. https://www.youtube.com/watch?v=q_ATY9gimOM

## Why It's Good For The Game

Make it less utterly debilitating to be blind!  It should be a huge inconvenience but shouldn't outright stop you from doing things.

## Changelog

:cl:
balance: blind people can now use health analyzers, atmospheric analyzers, and slime scanners!
balance: blind people can now use advanced camera consoles and shuttle designators.
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/camera_advanced.dm [hunk 1]
```diff
@@ -76,7 +76,7 @@
 	playsound(src, 'sound/machines/terminal_off.ogg', 25, FALSE)
 
 /obj/machinery/computer/camera_advanced/check_eye(mob/user)
-	if( (stat & (NOPOWER|BROKEN)) || (!Adjacent(user) && !user.has_unlimited_silicon_privilege) || user.eye_blind || user.incapacitated() )
+	if( (stat & (NOPOWER|BROKEN)) || (!Adjacent(user) && !user.has_unlimited_silicon_privilege) ||user.incapacitated() ) //monkestation edit
```
Comment (nednaZ) [inline code/game/machinery/computer/camera_advanced.dm:79]: I do have to question **how** a camera console can be used by the blind.
Audio feedback?
Computer: "There is a Clown and a Captain in view. The clown is slipping the Captain. The Captain is malding."

**Review by nednaZ — COMMENTED** [review]
Looks fine overall. Will test when I get a chance.
Are there any more random devices that are unable to be used by the blind?

File: code/game/machinery/computer/camera_advanced.dm (hunk 1 above)
Comment (KoboldCommando) [reply code/game/machinery/computer/camera_advanced.dm:79]: Yeah, what I pictured in my mind was effectively like that video I linked of a blind person browsing the internet.  Very fast text to speech that can also respond to inquiries.  Though talking purely game mechanics, blind people can use a number of different forms of camera or remote viewing, and enabling two more was easier and less disruptive than tracking down a ton more to disable!

**Comment by KoboldCommando** [comment]
> Looks fine overall. Will test when I get a chance. Are there any more random devices that are unable to be used by the blind?

I didn't notice any offhand, though there are several ways of checking for blindness so it's possible some slipped by me.  The main purpose of this was to make blind medic more playable though and I'd be happy to fix any others people point out!

File: code/game/machinery/computer/camera_advanced.dm [hunk 2]
```diff
@@ -76,7 +76,7 @@
 	playsound(src, 'sound/machines/terminal_off.ogg', 25, FALSE)
 
 /obj/machinery/computer/camera_advanced/check_eye(mob/user)
```
Comment (nednaZ) [inline code/game/machinery/computer/camera_advanced.dm:78]: As it stands, the advanced camera console shows 1 tile of view, while the regular camera console gives actual vision.

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -119,6 +119,19 @@ GENE SCANNER
 	user.visible_message("<span class='notice'>[user] analyzes [M]'s vitals.</span>", \
 						"<span class='notice'>You analyze [M]'s vitals.</span>")
 
+	//monkestation edit start
+	if(user.eye_blind) //based on the examinate verb in mob/mob.dm
```
Comment (nednaZ) [inline code/game/objects/items/devices/scanners.dm:123]: Radioactive scanners (Traitor item) instantly irradiates even if interrupted.
Not a big deal, but exists

File: code/game/objects/items/devices/scanners.dm [hunk 4]
```diff
@@ -473,9 +486,21 @@ GENE SCANNER
 /obj/item/analyzer/attack_self(mob/user)
 	add_fingerprint(user)
 
-	if(user.stat || user.eye_blind)
+	if(user.stat) //monkestation edit start
 		return
 
+	if(user.eye_blind)
```
Comment (nednaZ) [inline code/game/objects/items/devices/scanners.dm:492]: Ranged analyzers are instant if you target into the "fog of war"

File: code/modules/shuttle/shuttle_creation/shuttle_creator_console.dm [hunk 5]
```diff
@@ -11,7 +11,7 @@
 	var/datum/action/innate/shuttle_creator/airlock/airlock_action = new
 
 /obj/machinery/computer/camera_advanced/shuttle_creator/check_eye(mob/user)
-	if(user.eye_blind || user.incapacitated())
+	if(user.incapacitated()) //monkestation edit
```
Comment (nednaZ) [inline code/modules/shuttle/shuttle_creation/shuttle_creator_console.dm:14]: Works, but has the same issue as the advanced camera console.
One tile at a time.

**Review by nednaZ — APPROVED** [review]
Seems alright overall, I'll leave it up to you if you think the issues are inbalanced or not.

**Comment by nednaZ** [comment]
Inactive & has merge conflicts that were not addressed.

**Comment by KoboldCommando** [comment]
I thought this was already merged long ago?  What happened?

**Comment by nednaZ** [comment]
The issues that remained were never addressed, nor did you ever say if they were intended or not.

**Pull request closed without merging**